### PR TITLE
feat(single+multi-tenant): add support for single+multi-tenant.

### PR DESCRIPTION
**Pull request by cristhian-castaneda**
This PR enables Dinghy to run in single+multi-tenant mode, Dinghy will lookup for `mode` argument and depending on the value of this argument will start with that mode. 

Available modes:

single-tenant
multi-tenant

default: single-tenant

### CODE REVIEW

File: cmd/dinghy.go [hunk 1]
```diff
@@ -57,11 +58,17 @@ func newRedisOptions(redisOptions settings.Redis) *redis.Options {
...
-	config, err := settings.LoadSettings()
+	sourceConfiguration, err := settings.LoadSettings()
 	if err != nil {
 		log.Fatalf("failed to load configuration: %s", err.Error())
 	}
 
+	// We need to initialize the configuration for the start-up.
+	config, err := sourceConfiguration.LoadSetupSettings()
+	if err != nil {
+		log.Fatalf("%s", err.Error())
```
Comment (dogonthehorizon) [inline cmd/dinghy.go]: Is there more context we can provide here? Also I think we _should_ be able to do `log.Fatalf("%w", err)` to wrap the error.

File: cmd/dinghy.go [hunk 2]
```diff
@@ -109,14 +116,16 @@ func Setup() (*logr.Logger, *web.WebAPI) {
 	var logEventsClient logevents.LogEventsClient
 	var persitenceManager dinghyfile.DependencyManager
 	var persitenceManagerReadOnly dinghyfile.DependencyManager
+
 	// Full SQL mode
-	if config.SQL.Enabled  && !config.SQL.EventLogsOnly {
+	// TODO ensure this is correctly loaded when multi-tenant is used
```
Comment (dogonthehorizon) [inline cmd/dinghy.go]: Is this still a todo? Is there a test we can write to validate this?

File: cmd/dinghy.go [hunk 3]
```diff
@@ -147,52 +155,52 @@ func Setup() (*logr.Logger, *web.WebAPI) {
...
 		persitenceManager = redisClient
 		persitenceManagerReadOnly = &redisClientReadOnly
 
 	} else {
 		// Redis mode
-		redisClient := cache.NewRedisCache(newRedisOptions(config.Redis), log, ctx, stop, true)
+		redisClient := cache.NewRedisCache(newRedisOptions(config.SpinnakerSupplied.Redis), log, ctx, stop, true)
 		if _, err := redisClient.Client.Ping().Result(); err != nil {
-			log.Fatalf("Redis Server at %s could not be contacted: %v", config.Redis.BaseURL, err)
+			log.Errorf("Redis Server at %s could not be contacted: %v", config.SpinnakerSupplied.Redis.BaseURL, err)
```
Comment (dogonthehorizon) [inline cmd/dinghy.go]: Why remove the Fatal here? My understanding of the branching now is that we'll revert to Redis if SQL isn't available, which would imply that persistence won't work correctly. Is that intended?

Comment (cristhian-castaneda) [reply cmd/dinghy.go]: That’s a good point, I just updated it.

That was one of the first changes I did, it made sense at that moment haha.

File: cmd/dinghy.go (hunk 2 above)
Comment (cristhian-castaneda) [reply cmd/dinghy.go]: Done ✅ !!

File: cmd/dinghy.go (hunk 1 above)
Comment (cristhian-castaneda) [reply cmd/dinghy.go]: Done ✅ !!

**Review by dogonthehorizon — APPROVED** [review]

**Pull request merged**